app/src/main/java/com/example/characters/fragment/UserListFragment.kt

```diff
@@ -0,0 +1,159 @@
+                UserListFragmentDirections.toUserDetails(
+                    it.id,
+                    it.name
+                )
+            )
+        }
+    }
+
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        loadingUserRetrofit()
+
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        return FragmentListUsersBinding.inflate(inflater, container, false)
+            .also { _binding = it }
+            .root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        with(binding) {
+
+            swipeRefresh.setOnRefreshListener {
+                iSLoadingCount = false
+                loadingUserRetrofit()
+                swipeRefresh.isRefreshing = false
+            }
+
+            val layoutManager = LinearLayoutManager(requireContext())
+
+            listUsers.adapter = adapter
+            listUsers.layoutManager = layoutManager
+            listUsers.addDecorationUser(bottomDecorator = FIFTEEN_DP)
+
+            listUsers.addPaginationScrollListener(layoutManager, 1) {
+                if (!isLoading) {
+                    loadingUserRetrofit()
+                }
+            }
+        }
+    }
+
+    private fun loadingUserRetrofit() {
+        retrofitService = RetrofitService.loadingRetrofitService().getUsers()
+        isLoading = true
+        retrofitService?.enqueue(object : Callback<List<UserLoading.User>> {
+            override fun onResponse(
+                call: Call<List<UserLoading.User>>,
+                response: Response<List<UserLoading.User>>
+            ) {
+                if (response.isSuccessful) {
+                    val user = response.body() ?: return
+                    if (iSLoadingCount) {
+                        val currentList = adapter.currentList.toList().dropLast(1)
+                        val resultList = currentList
+                            .plus(user)
+                            .plus(UserLoading.Loading)
+                        adapter.submitList(resultList)
+                        isLoading = false
+                    } else {
+                        adapter.submitList(user + UserLoading.Loading)
+                        iSLoadingCount = true
+                        isLoading = false
+                    }
+                } else {
+                    Toast.makeText(
+                        requireContext(),
+                        HttpException(response).message(),
+                        Toast.LENGTH_LONG
+                    )
+                        .show()
+                }
+
+                retrofitService = null
+            }
+
+            override fun onFailure(call: Call<List<UserLoading.User>>, t: Throwable) {
+                if (call.isCanceled && t == SocketException()) {
```

Comment:
проверка на тип идет через слово `is`, в твоем случае ты создаешь новый объект исключения и надеешься что они будут равны по equals
```suggestion
                if (call.isCanceled && t is SocketException) {
```